packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -0,0 +1,439 @@
+            ? isBillingAddressSameData.cart.isBillingAddressSame
+            : true;
+
+        let billingAddress = {};
+        if (billingAddressData) {
+            if (billingAddressData.cart.billingAddress) {
+                const {
+                    // eslint-disable-next-line no-unused-vars
+                    __typename,
+                    ...rawBillingAddress
+                } = billingAddressData.cart.billingAddress;
+                billingAddress = mapAddressData(rawBillingAddress);
+            }
+        }
+
+        return { isBillingAddressSame, ...billingAddress };
+    }, [isBillingAddressSameData, billingAddressData]);
+
+    const errors = useMemo(() => {
+        const errors = [];
+
+        if (ccMutationErrors) {
+            ccMutationErrors.graphQLErrors.forEach(({ message }) => {
+                errors.push(message);
+            });
+        }
+        if (billingAddressMutationErrors) {
+            billingAddressMutationErrors.graphQLErrors.forEach(
+                ({ message }) => {
+                    errors.push(message);
+                }
+            );
+        }
+
+        return errors;
+    }, [ccMutationErrors, billingAddressMutationErrors]);
+
+    /**
+     * Helpers
+     */
+
+    const setIsBillingAddressSameInCache = useCallback(() => {
+        client.writeQuery({
+            query: getIsBillingAddressSameQuery,
+            data: {
+                cart: {
+                    __typename: 'Cart',
+                    id: cartId,
+                    isBillingAddressSame
+                }
+            }
+        });
+    }, [client, cartId, getIsBillingAddressSameQuery, isBillingAddressSame]);
+
+    const setShippingAddressAsBillingAddress = useCallback(() => {
+        const shippingAddress = shippingAddressData
+            ? mapAddressData(shippingAddressData.cart.shippingAddresses[0])
+            : {};
+        const {
+            firstName = '',
+            lastName = '',
+            country = '',
+            street1 = '',
+            street2 = '',
+            city = '',
+            state = '',
+            postalCode = '',
+            phoneNumber = ''
+        } = shippingAddress;
+
+        updateBillingAddress({
+            variables: {
+                cartId,
+                firstName,
+                lastName,
+                street1,
+                street2,
+                city,
+                state,
+                postalCode,
+                country,
+                phoneNumber,
+                sameAsShipping: true
+            }
+        });
+    }, [updateBillingAddress, shippingAddressData, cartId]);
+
+    const setBillingAddress = useCallback(() => {
+        const {
+            firstName = '',
+            lastName = '',
+            country = '',
+            street1 = '',
+            street2 = '',
+            city = '',
+            state = '',
+            postalCode = '',
+            phoneNumber = ''
+        } = formState.values;
+
```

Comment:
Wouldn't this end up suppressing any errors from GraphQL? Is that the behavior we want?

Reply:
You are right, Ill change this.